scripts/generate_certs.py

```diff
@@ -0,0 +1,156 @@
+        output += '  ' + '  '.join(data.splitlines(keepends=True)[1:-2])
+        output = "".join([(line + (77-len(line))*" " + "\\\n") for line in output.splitlines()])
+        output += "}\n"
+        return output
+
+    @staticmethod
+    def type_macro_string(name, filename):
+        """ Generates code snipper to be added
+        name: name of macro
+        filename: where value of macro stored
+        """
+        output = "#define " + name + "\n"
+        with open(filename) as f:
+            for line in f.read().splitlines()[:-1]:
+                output += "    \"" + line + "\\r\\n\"\n"
+            output = output[:-1] + "\n"
+        output = "".join([(line + (75-len(line))*" " + "\\\n") for line in output.splitlines()])
+        with open(filename) as f:
+            output += "    \"" + f.read().splitlines()[-1] + "\\r\\n\"\n"
+        return output
+
+    # Extract key or certificate from file.
+    def extract_key(self, enc, typ, name, filename):
+        """ Depending on encoding and type it calls appriopriate
+        function
+
+        enc: encoding
+        typ: type
+        name: name of variable/macro
+        filename: where data is
+        """
+        if enc not in ("string", "binary"):
+            print("choose either string encoding or binary encoding")
+            sys.exit(1)
+        if typ not in ("macro", "variable"):
+            print("choose either macro or variable as a type")
+            sys.exit(1)
+
+        output = ""
+        if typ == "variable":
+            if enc == "binary":
+                output = self.type_var_bin(name, filename)
+            elif enc == "string":
+                output = self.type_var_string(name, filename)
+        elif typ == "macro":
+            if enc == "binary":
+                output = self.type_macro_bin(name, filename)
+            elif enc == "string":
+                output = self.type_macro_string(name, filename)
+        return output
+
+    def update(self):
+        """ Updates cert file using data given in comment
+        sections. Location of certs.c file is hardcoded
+        """
+        tempf = tempfile.TemporaryFile(mode="w+")
+        with open(self.certs, "r+") as old_f, open(tempf.name, "w+") as tmp:
+            line = old_f.readline()
+            while line:
+                if re.fullmatch(r"^/\*\s*BEGIN FILE.*\*/$\n", line):
+                    tmp.write(line)
+                    args = line.split(" ")[3:7]
+                    add = self.extract_key(args[0], args[1], args[2], args[3])
+                    tmp.write(add)
+                    line = old_f.readline()
+                    while not re.fullmatch(r"^/\*\s*END FILE\s*\*/$\n", line):
+                        line = old_f.readline()
+                tmp.write(line)
+                line = old_f.readline()
+            tmp.seek(0)
+            old_f.truncate(0)
+            final = tmp.read()
+            old_f.seek(0)
+            old_f.write(final)
```

Comment:
final not really required 
```suggestion
            old_f.write(tmp.read())
```

Reply:
changed code to not use final variable